docs/rfcs/rfc-035-authorization.md

```diff
@@ -0,0 +1,759 @@
+    name_SHORTEST_EQUAL: Int @deprecated(reason: \\"Please use the explicit _LENGTH version for string aggregation.\\")
+    name_SHORTEST_GT: Int @deprecated(reason: \\"Please use the explicit _LENGTH version for string aggregation.\\")
+    name_SHORTEST_GTE: Int @deprecated(reason: \\"Please use the explicit _LENGTH version for string aggregation.\\")
+    name_SHORTEST_LENGTH_EQUAL: Int
+    name_SHORTEST_LENGTH_GT: Int
+    name_SHORTEST_LENGTH_GTE: Int
+    name_SHORTEST_LENGTH_LT: Int
+    name_SHORTEST_LENGTH_LTE: Int
+    name_SHORTEST_LT: Int @deprecated(reason: \\"Please use the explicit _LENGTH version for string aggregation.\\")
+    name_SHORTEST_LTE: Int @deprecated(reason: \\"Please use the explicit _LENGTH version for string aggregation.\\")
+}
+
+input UserWhere {
+    AND: [UserWhere!]
+    NOT: UserWhere
+    OR: [UserWhere!]
+    author: UserWhere
+    authorAggregate: UserAuthorAggregateInput
+    authorConnection: UserAuthorConnectionWhere
+    authorConnection_NOT: UserAuthorConnectionWhere
+    author_NOT: UserWhere
+    id: ID
+    id_CONTAINS: ID
+    id_ENDS_WITH: ID
+    id_IN: [ID!]
+    id_NOT: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_CONTAINS: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_ENDS_WITH: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_IN: [ID!] @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_NOT_STARTS_WITH: ID @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    id_STARTS_WITH: ID
+    name: String
+    name_CONTAINS: String
+    name_ENDS_WITH: String
+    name_IN: [String!]
+    name_NOT: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_CONTAINS: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_ENDS_WITH: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_IN: [String!] @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_NOT_STARTS_WITH: String @deprecated(reason: \\"Negation filters will be deprecated, use the NOT operator to achieve the same behavior\\")
+    name_STARTS_WITH: String
+}
+```
+
+#### JWT Payload schema
+
+The generated types for the directive will depends on `jwtPayload` when instantiating the feature. Given the following:
+
+```ts
+new Neo4jGraphQL({
+  typeDefs,
+  features: {
+    authorization: {
+      secret: "secret",
+      jwtPayload: {
```

Comment:
Liam also mentioned this on the last RFC... I don't think I like enforcing users into using the type name `JWTPayload`, however, I would be open to introducing a labelling directive:

```gql
type SomeOtherType @jwtPayload {
  sub: String!
  roles: [String!]!
}
```

Thoughts?

Reply:
yees!

Reply:
We would check for only one instance of `@jwtPayload` in the entire schema. I don't think something on schema would work, let's take advantage of the type safety of GraphQL!

Reply:
I like it! potentially could be beneficial for some users to have not a single instance of `@jwtPayload`, if we apply the directive on an interface and we allow only one interface to be defined with the `@jwtPayload`, then it will be easier to support multiple `jwtPayload` implementations if requested, as we could reuse the interface filtering syntax.